### PR TITLE
Fix InvokeSimd simd_size and call_vadd_1d_loop_naive tests

### PR DESCRIPTION
**Pull request by fveselov**

### CODE REVIEW

File: SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp [hunk 1]
```diff
@@ -6,15 +6,6 @@
 #include <iostream>
 #include <type_traits>
 
-/* Subgroup size attribute is optional
- * In case it is absent compiler decides what subgroup size to use
- */
-#ifdef IMPL_SUBGROUP
```
Comment (sarnex) [inline SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp:12]: Sorry why are we changing this? Do we not want to test it without an explicit subgroup size?

File: SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  passed &= test<16, 8>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (8) does not match size of invoke_simd return type (16){{.*}}
+  passed &= test<32, 8>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (8) does not match size of invoke_simd return type (32){{.*}}
+
+  // simd_size 16
+  passed &= test<4, 16>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (16) does not match size of invoke_simd return type (4){{.*}}
+  passed &= test<8, 16>(q);
+  // CHECK: {{.*}}error: Kernel subgroup size (16) does not match size of invoke_simd return type (8){{.*}}
```
Comment (sarnex) [inline SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp:40]: Were these originally expected to pass with no error?

Comment (v-klochkov) [reply SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp:40]: invoke_simd SPEC was not clear enough.
Actually it requires that If the target of invoke_simd has any simd::experimental::simd<T,N> argument or return, then N must be equal to the sub-group-size in SYCL context.

File: SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// TODO: enable when Jira ticket resolved
+// XFAIL: gpu
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+/* Tests invoke_simd support in the compiler/headers
+ * Test checks an error message when subgroup size mismatches invoke_simd
+ * return type (simd<T, M> or simd_mask<T, M>) size
```
Comment (v-klochkov) [inline SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp]: Minor comment: it verifies not only 'return', but also 'arguments' of invoke_simd target function.
```suggestion
 * The test checks that compiler emits a meaningful and user friendly error message
 * when the target function of invoke_simd has arguments or returns of simd<T,N> or
 * simd_mask<T,N> type and N is not equal to subgroup size.
```

File: SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp (hunk 1 above)
Comment (fveselov) [reply SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp:12]: It turned out that cases where subgroup size does not match size of invoke_simd return type or uniform argument type are not allowed.
At this point main purpose of most simd_size cases is to make sure that the compiler reports an error when an invalid configuration is used.
All tests that expect to compile and execute successfully (simd8.cpp, simd16.cpp and simd32.cpp) now contain only one case where the sizes are equal.
This configuration is widely used in other InvokeSimd tests, that have implicit subgroup version.

File: SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp (hunk 2 above)
Comment (sarnex) [reply SYCL/InvokeSimd/Spec/simd_size/simd_mismatch_error.cpp:40]: thanks!

File: SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp (hunk 1 above)
Comment (sarnex) [reply SYCL/InvokeSimd/Spec/simd_size/Inputs/common.hpp:12]: thanks!

**Review by sarnex — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**